### PR TITLE
format_datetime: using 'S...S' format is incorrect

### PR DESCRIPTION
**Pull request by iamshubh22**
Fixes https://github.com/python-babel/babel/issues/350

### CODE REVIEW

File: babel/dates.py [hunk 1]
```diff
@@ -1287,8 +1287,13 @@ def format_period(self, char):
         return get_period_names(locale=self.locale)[period]
 
     def format_frac_seconds(self, num):
-        value = str(self.value.microsecond)
-        return self.format(round(float('.%s' % value), num) * 10**num, num)
+        """ Return fractional seconds.
+
+        Rounds the time's microseconds to the precision given by the number \
```
Comment (akx) [inline babel/dates.py]: No need to use the backslash for line continuations in docstrings :)

Comment (iamshubh22) [inline babel/dates.py]: @akx Ok. Got it :smile:

**Comment by iamshubh22** [comment]
@akx TravisCI is failing for pypy for Linux. Why so?

**Comment by akx** [comment]
@iamshubh22 Ah, nothing to worry about if I'm guessing right. I restarted that particular build now.

To be exact, the problem is that occasionally (very seldom though) time manages to roll over a second during a test, so when a test expects a file to say `the time right now` (whatever it is), it may be `the time right now + 1 second`, and that makes the test fail.

**Comment by akx** [comment]
This looks good! 

Thank you for the patch, @iamshubh22! :trophy:

**Comment by iamshubh22** [comment]
@akx Just one doubt. What does this Code coverage tells??

**Comment by akx** [comment]
@iamshubh22:

* *codecov/patch* - 100.00% of diff hit (target 80.00%)
  * This means that all of the lines of code you've touched in the PR are also being tested.
* *codecov/project* - 89.66% remains the same compared to e1ce012
  * This means that coverage doesn't change -- as is reasonable. The previous version was 2 lines of code, the new version is 2 lines of code too.

**Pull request merged**